### DISCUSSION TITLE
About the `+` appending to the folder name

### DISCUSSION THREAD

**Discussion by HuyAms** [kiliman/remix-flat-routes#108] [Q&A] [answered]
Hi, 

I am checking this document about the `+` appending to the folder name
https://github.com/kiliman/remix-flat-routes?tab=readme-ov-file#nested-folders-with-flat-files-convention--new-in-v051

Why the `project` folder doesn't have the `+` but the `users` does? 
![CleanShot 2024-02-14 at 11 23 59@2x](https://github.com/kiliman/remix-flat-routes/assets/26871154/4ad0f40c-2c44-4e68-85f2-794852dd0792)

When checking the routes, it looks quite the same to me

```
 <Route path="project" file="routes-hybrid-files/project/_layout.tsx">
      <Route
        path="parent/child"
        file="routes-hybrid-files/project/parent.child/index.tsx"
      >
        <Route
          path="grandchild"
          file="routes-hybrid-files/project/parent.child.grandchild/index.tsx"
        />
     </Route>
</Route>


<Route path="users" file="routes-hybrid-files/users+/_layout.tsx">
      <Route path=":userId" file="routes-hybrid-files/users+/$userId.tsx" />
      <Route
        path=":userId/edit"
        file="routes-hybrid-files/users+/$userId_.edit.tsx"
      />
      <Route index file="routes-hybrid-files/users+/index.tsx" />
</Route>
   ```

Could anyone kindly explain it to me?

Thank you so much!

**Comment by kiliman** [comment, marked as answer]
The `project` folder _should_ have a `+`, since it's nested, not colocated.

The reason it currently works is due to a **bug** where it treats _interior_ `/` as though it were `.`. Only the final _naked_ folder is treated as a colocated folder.

`remix-flat-routes` converts the route into the `v2` flat routes convention before processing. This is to ensure that it will always be _backwards_ compatible with the official Remix convention.

That is why `routes/project/parent.child.grandchild/styles.css` is treated as `routes/project.parent.child.grandchild/styles.css` and is considered a colocated file.

Since this **bug** is probably _inadvertently_ depended on, once it's fixed I will add a config option to opt-in to it.
